WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsDialogFragment.kt

```diff
@@ -0,0 +1,399 @@
+        innerView.orientation = LinearLayout.VERTICAL
+
+        if (mShouldDisplayMainSwitch) {
+            val dividerView = View(context)
+            val dividerHeight = context.resources.getDimensionPixelSize(
+                R.dimen.notifications_settings_dialog_divider_height
+            )
+            dividerView.background = context.getDrawableFromAttribute(android.R.attr.listDivider)
+            dividerView.layoutParams = ViewGroup.LayoutParams(ActionBar.LayoutParams.MATCH_PARENT, dividerHeight)
+            innerView.addView(dividerView)
+        } else {
+            val spacerView = View(context)
+            val spacerHeight = context.resources.getDimensionPixelSize(R.dimen.margin_medium)
+            spacerView.layoutParams = ViewGroup.LayoutParams(ActionBar.LayoutParams.MATCH_PARENT, spacerHeight)
+            innerView.addView(spacerView)
+        }
+
+        mDisabledView = View.inflate(context, R.layout.notifications_tab_disabled_text_layout, null)
+        mDisabledView?.layoutParams = ViewGroup.LayoutParams(
+            ActionBar.LayoutParams.MATCH_PARENT,
+            ActionBar.LayoutParams.WRAP_CONTENT)
+
+        mOptionsView = LinearLayout(context)
+        mOptionsView?.layoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT)
+        mOptionsView?.orientation = LinearLayout.VERTICAL
+        innerView.addView(mDisabledView)
+        innerView.addView(mOptionsView)
+        outerView.addView(innerView)
+        configureLayoutForView(mOptionsView!!)
+
+        val builder: AlertDialog.Builder = MaterialAlertDialogBuilder(requireActivity()).apply {
+            setTitle(title)
+            setPositiveButton(android.R.string.ok, this@NotificationsSettingsDialogFragment)
+            setNegativeButton(R.string.cancel, this@NotificationsSettingsDialogFragment)
+            setView(layout)
+        }
+
+        if (mShouldDisplayMainSwitch) {
+            setupTitleViewWithMainSwitch(outerView)
+            if (mTitleViewWithMainSwitch == null)
+                AppLog.e(AppLog.T.NOTIFS, "Main switch enabled but layout not set")
+            else
+                builder.setCustomTitle(mTitleViewWithMainSwitch)
+        }
+
+        return builder.create()
+    }
+
+    private fun setupTitleViewWithMainSwitch(view: View) {
+        when (this.channel) {
+            NotificationsSettings.Channel.BLOGS -> if (this.type == NotificationsSettings.Type.TIMELINE) {
+                mTitleViewWithMainSwitch = layoutInflater
+                    .inflate(R.layout.notifications_tab_for_blog_title_layout, view as ViewGroup, false) as ViewGroup
+            }
+
+            NotificationsSettings.Channel.OTHER, NotificationsSettings.Channel.WPCOM -> {}
+        }
+        if (mTitleViewWithMainSwitch != null) {
+            val titleView = mTitleViewWithMainSwitch!!.findViewById<TextView>(R.id.title)
```

Comment:
One other generic issue not caught by the tools is the use of [not-null assertion operator](https://kotlinlang.org/docs/null-safety.html#the-operator) (!!) in many places in the introduced code. Though this may be introduced when converting code to Kotlin we should try to avoid it.
Since this specific case involves a `View` we could use a `lateinit` variable instead of a nullable one and not have the need for a `!!`. This can be applied in all interface objects used. Wdyt?
When using a nullable object is not avoidable it is preferred to use the [safe call operator `?.`](https://kotlinlang.org/docs/null-safety.html#safe-calls) or a simple null check.
You can find some more information on null safety in Kotlin [here](https://kotlinlang.org/docs/null-safety.html).